src/features/utils/instrument-base.test.js

```diff
@@ -0,0 +1,187 @@
+  expect(mockAggregate).toHaveBeenCalledWith(agentIdentifier, aggregator, aggregateArgs)
+})
+
+test('should import the session manager and replay aggregate for new session', async () => {
+  jest.mocked(getConfigurationValue).mockReturnValue(true)
+  jest.mocked(setupAgentSession).mockReturnValue({
+    isNew: true
+  })
+
+  const instrument = new InstrumentBase(agentIdentifier, aggregator, FEATURE_NAMES.sessionReplay)
+  const aggregateArgs = { [faker.datatype.uuid()]: faker.lorem.sentence() }
+  instrument.importAggregator(aggregateArgs)
+
+  const windowLoadCallback = jest.mocked(onWindowLoad).mock.calls[0][0]
+  await windowLoadCallback()
+
+  expect(getConfigurationValue).toHaveBeenCalledWith(agentIdentifier, 'privacy.cookies_enabled')
+  expect(setupAgentSession).toHaveBeenCalledWith(agentIdentifier)
+  expect(lazyFeatureLoader).toHaveBeenCalledWith(FEATURE_NAMES.sessionReplay, 'aggregate')
+  expect(mockAggregate).toHaveBeenCalledWith(agentIdentifier, aggregator, aggregateArgs)
+})
+
+test('should import the session manager and replay aggregate when a recording is active', async () => {
+  jest.mocked(getConfigurationValue).mockReturnValue(true)
+  jest.mocked(setupAgentSession).mockReturnValue({
+    isNew: false,
+    state: {
+      sessionReplayActive: true
+    }
+  })
+
+  const instrument = new InstrumentBase(agentIdentifier, aggregator, FEATURE_NAMES.sessionReplay)
+  const aggregateArgs = { [faker.datatype.uuid()]: faker.lorem.sentence() }
+  instrument.importAggregator(aggregateArgs)
+
+  const windowLoadCallback = jest.mocked(onWindowLoad).mock.calls[0][0]
+  await windowLoadCallback()
+
+  expect(getConfigurationValue).toHaveBeenCalledWith(agentIdentifier, 'privacy.cookies_enabled')
+  expect(setupAgentSession).toHaveBeenCalledWith(agentIdentifier)
+  expect(lazyFeatureLoader).toHaveBeenCalledWith(FEATURE_NAMES.sessionReplay, 'aggregate')
+  expect(mockAggregate).toHaveBeenCalledWith(agentIdentifier, aggregator, aggregateArgs)
+})
+
+test('should not import session aggregate when session is not new and a recording is not active', async () => {
+  jest.mocked(getConfigurationValue).mockReturnValue(true)
+  jest.mocked(setupAgentSession).mockReturnValue({
+    isNew: false,
+    state: {
+      sessionReplayActive: false
+    }
+  })
+
+  const instrument = new InstrumentBase(agentIdentifier, aggregator, FEATURE_NAMES.sessionReplay)
+  const aggregateArgs = { [faker.datatype.uuid()]: faker.lorem.sentence() }
+  instrument.importAggregator(aggregateArgs)
+
+  const windowLoadCallback = jest.mocked(onWindowLoad).mock.calls[0][0]
+  await windowLoadCallback()
+
+  expect(getConfigurationValue).toHaveBeenCalledWith(agentIdentifier, 'privacy.cookies_enabled')
+  expect(setupAgentSession).toHaveBeenCalledWith(agentIdentifier)
+  expect(drain).toHaveBeenCalledWith(agentIdentifier, FEATURE_NAMES.sessionReplay)
+  expect(lazyFeatureLoader).not.toHaveBeenCalled()
+  expect(mockAggregate).not.toHaveBeenCalled()
+})
+
```

Comment:
Given that the "warn-don't-error" behavior is important, maybe we need a test like "should catch and warn when the session manager fails to import"—applying to the catch condition on `instrument-base.js:53`.

Reply:
I was not able to find a way to `spyOn` the `import` function without it causing issues but maybe I can try again.

Reply:
Yeah, that's why I left you a comment instead of a code suggestion! I don't think it's worth the effort from a cost-benefit perspective. Resolving this conversation.